### ISSUE TITLE
`-Mode` parameter for `Set-StrictMode`

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#16388]
### Summary of the new feature / enhancement

Setting [`Set-StrictMode`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-strictmode) to a specific `-Version` level automatically implies that *all prior* prohibitors are enabled. Meaning that there is no way just select for a recent prohibitor (e.g. "*out of bounds or unresolvable array indexes*") without also selecting a earlier prohibitor (e.g. "*references to uninitialized variables*").

A `-Modes` parameter that supports certain prohibit flags, e.g.:
```PowerShell
[Flags()] enum StrictModes {
    ProhibitReferencesToUninitializedVariables = 1
    ProhibitReferencesToNonExistentProperties = 2
    ProhibitUseOfCallingMethodSyntax = 4
    ProhibitUnresolvableArrayIndexes = 8
    ProhibitOmittingWriteOutput = 16
}
```
will avoid that.
The lack of selective prohibitors also holds back the future extensions of the `Set-StrictMode` cmdlet as e.g. [`#15781`Strict Write-Output](https://github.com/PowerShell/PowerShell/issues/15781).

**Comment by iRon7** [comment]
Also possible (false positive) bugs in specific `Set-StrictMode` modes as in [`#16427` Set-StrictMode -Version Latest: False positive for references to non-existent properties of an object](https://github.com/PowerShell/PowerShell/issues/16427) could be excluded/suppressed while still being able to as strict as possible with the rest of the prohibitors, using a command as e.g.: `Set-StrictMode -Mode 29`

**Comment by iRon7** [comment]
@mklement0,

> Even if lexical scoping were to be introduced, the dynamic scoping of the existing strict modes would have to be retained for backward compatibility, which would invite confusion and subtle bugs.

A different way to implement lexical scoping is by excluding using [SuppressMessageAttribute Class](https://docs.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute) in the cmdlet binding similar to the approach of PSScriptAnalyzer.

**Comment by iRon7** [comment]
I have closed this issue in favor of issue #20432
